src/android/CodePush.java

```diff
@@ -275,9 +275,19 @@ private void clearDeploymentsIfBinaryUpdated() {
         if (deployedPackageMetadata != null) {
             String deployedPackageTimeStamp = deployedPackageMetadata.nativeBuildTime;
             long nativeBuildTime = Utilities.getApkBuildTime(this.cordova.getActivity());
-            if (nativeBuildTime != -1) {
+
+            String deployedPackageVersion = deployedPackageMetadata.appVersion;
+            String applicationVersion = null;
+            try {
+                applicationVersion = Utilities.getAppVersionName(this.cordova.getActivity());
+            } catch (PackageManager.NameNotFoundException e) {
+                e.printStackTrace();
```

Comment:
Yes, I thought about it too, I believe that if we don't have version defined than the app is not correct at all.